### ISSUE TITLE
Convert to py.test?

### ISSUE THREAD

**Issue by bmcfee** [swcarpentry/python-testing#2]
Since nose is effectively [abandoned](http://nose.readthedocs.io/en/latest/), should this lesson be converted to use [py.test](http://docs.pytest.org/en/latest/) instead?

Little would functionally change, since py.test can run tests written for nose without any problems.  The main differences would be in fixture generation and marking failure cases.

FWIW, I've been porting test suites for some of my projects from nose to py.test, and it's pretty easy.

**Comment by jiffyclub** [comment]
Yes we should do this. I did this downstream for the SciPy workshop and it was not too bad: https://github.com/katyhuff/2016-07-11-scipy/pull/1

**Comment by bmcfee** [comment]
@jiffyclub Are those changes easy enough to merge upstream, or should we redo it directly here?

**Comment by jiffyclub** [comment]
It might be kind of challenging to replay the changes here because the repo organization is different. Otherwise I think everything is the same so the changes would port over easily.

**Comment by bmcfee** [comment]
Sounds good.  I can pick this one up and make it my PR for satisfying instructor training.
